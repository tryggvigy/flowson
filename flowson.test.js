const flowson = require('./flowson')
const pkg = require('./package.json')

describe('flowson', () => {
  it('handles empty objects', () => {
    expect(
      flowson('config.json', '{}')
    ).toEqual('// @flow\nexport type configT = {}')
  })

  it('transforms json as expected', () => {
    expect(flowson(
      'config.json', 
      `{
        "str": "cake",
        "number": 1,
        "bool": false,
        "arr": [1, true, "foo", {}, [1, {"foo": "bar"}]],
        "nestedObj": {
          "arr": [1, true, "foo", {}, [1, {"foo": "bar"}]],
          "foo": "bar",
          "Baz-foo": {"baz_": 1}
        }
      }`
    )).toMatchSnapshot()
  })
  
  it('can set indentation and typename', () => {
    expect(flowson(
      'config.json', 
      `{
        "str": "cake",
        "foo": [1,2],
        "nestedObj": {
          "foo": "bar"
        }
      }`,
      { indent: 4, typename: 'stupidNameT' }
    )).toMatchSnapshot()
  })
  
})
