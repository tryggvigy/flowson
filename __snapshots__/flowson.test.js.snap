// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flowson can set indentation and typename 1`] = `
"// @flow
export type configT = {
    \\"str\\": \\"cake\\",
    \\"foo\\": [
        1,
        2
    ],
    \\"nestedObj\\": {
        \\"foo\\": \\"bar\\"
    }
}"
`;

exports[`flowson transforms json as expected 1`] = `
"// @flow
export type configT = {
  \\"str\\": \\"cake\\",
  \\"number\\": 1,
  \\"bool\\": false,
  \\"arr\\": [
    1,
    true,
    \\"foo\\",
    {},
    [
      1,
      {
        \\"foo\\": \\"bar\\"
      }
    ]
  ],
  \\"nestedObj\\": {
    \\"arr\\": [
      1,
      true,
      \\"foo\\",
      {},
      [
        1,
        {
          \\"foo\\": \\"bar\\"
        }
      ]
    ],
    \\"foo\\": \\"bar\\",
    \\"Baz-foo\\": {
      \\"baz_\\": 1
    }
  }
}"
`;
